Dapper_Crud/Controllers/SuperHeroController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Dapper;
+using Microsoft.AspNetCore.Mvc;
+using System.Data.SqlClient;
+
+namespace Dapper_Crud.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class SuperHeroController : ControllerBase
+    {
+        private IConfiguration configuration;
+        public SuperHeroController(IConfiguration config)
+        {
+            configuration = config;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<List<SuperHero>>> GetAllSuperHeros()
+        {
+            using var connection = new SqlConnection(configuration.GetConnectionString("DefaultConnection"));
+            var Heros = await connection.QueryAsync<SuperHero>("SELECT * From superheroes");
```

Comment:
create seperate infrastructure folder and put all sql operations thr. In controller inject the infrastructure

Reply:
Dependency Injection Implemented